File: DevExpress-Demo/VB/BarsDemo.Wpf/Controls/RootView.vb
Imports DevExpress.Mvvm.Native
Imports DevExpress.Xpf.Bars
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Data

Namespace BarsDemo

    Public Class RootView
        Inherits ContentControl

        Shared Sub New()
            Call DefaultStyleKeyProperty.OverrideMetadata(GetType(RootView), New FrameworkPropertyMetadata(GetType(RootView)))
        End Sub

        Protected Overrides Sub OnContentChanged(ByVal oldContent As Object, ByVal newContent As Object)
            MyBase.OnContentChanged(oldContent, newContent)
            TryCast(newContent, DependencyObject).[Do](Sub(x) BindingOperations.SetBinding(x, MergingProperties.ToolBarMergeStyleProperty, New Binding() With {.Path = New PropertyPath(MergingProperties.ToolBarMergeStyleProperty), .Source = Me}))
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/CS/BarsDemo.Wpf/Controls/RootView.cs
using DevExpress.Mvvm.Native;
using DevExpress.Xpf.Bars;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Data;
namespace BarsDemo {
    public class RootView : ContentControl {
        static RootView() {
            DefaultStyleKeyProperty.OverrideMetadata(typeof(RootView), new FrameworkPropertyMetadata(typeof(RootView)));
        }
        protected override void OnContentChanged(object oldContent, object newContent) {
            base.OnContentChanged(oldContent, newContent);
            (newContent as DependencyObject)
                .Do(x => BindingOperations.SetBinding(
                    x,
                    MergingProperties.ToolBarMergeStyleProperty,
                    new Binding() {
                        Path = new PropertyPath(MergingProperties.ToolBarMergeStyleProperty),
                        Source = this
                    }));
        }
    }
}
